allennlp/semparse/worlds/wikitables_variable_free_world.py

```diff
@@ -0,0 +1,174 @@
+from allennlp.semparse.contexts import TableQuestionKnowledgeGraph
+
+
+class WikiTablesVariableFreeWorld(World):
+    """
+    World representation for the WikitableQuestions domain with the variable-free language used in
+    the paper from Liang et al. (2018).
+
+    Parameters
+    ----------
+    table_graph : ``TableQuestionKnowledgeGraph``
+        Context associated with this world.
+    """
+    # When we're converting from logical forms to action sequences, this set tells us which
+    # functions in the logical form are curried functions, and how many arguments the function
+    # actually takes.  This is necessary because NLTK curries all multi-argument functions to a
+    # series of one-argument function applications.  See `world._get_transitions` for more info.
+    curried_functions = {
+            types.SELECT_TYPE: 2,
+            types.ROW_FILTER_WITH_COLUMN: 2,
+            types.ROW_NUM_OP: 2,
+            types.ROW_FILTER_WITH_COLUMN_AND_NUMBER: 3,
+            types.ROW_FILTER_WITH_COLUMN_AND_DATE: 3,
+            types.ROW_FILTER_WITH_COLUMN_AND_STRING: 3,
+            types.NUM_DIFF_WITH_COLUMN: 3,
+            }
+
+    def __init__(self, table_graph: TableQuestionKnowledgeGraph) -> None:
+        super().__init__(constant_type_prefixes={"string": types.STRING_TYPE,
+                                                 "num": types.NUMBER_TYPE},
+                         global_type_signatures=types.COMMON_TYPE_SIGNATURE,
+                         global_name_mapping=types.COMMON_NAME_MAPPING)
+        self.table_graph = table_graph
+
+        # For every new Sempre column name seen, we update this counter to map it to a new NLTK name.
+        self._column_counter = 0
+
+        # This adds all of the cell and column names to our local name mapping.
+        for entity in table_graph.entities:
+            self._map_name(entity, keep_mapping=True)
+
+        self._entity_set = set(table_graph.entities)
+        self.terminal_productions: Dict[str, str] = {}
+        for entity in self._entity_set:
+            mapped_name = self.local_name_mapping[entity]
+            signature = self.local_type_signatures[mapped_name]
+            self.terminal_productions[entity] = f"{signature} -> {entity}"
+
+        for predicate, mapped_name in self.global_name_mapping.items():
+            if mapped_name in self.global_type_signatures:
+                signature = self.global_type_signatures[mapped_name]
+                self.terminal_productions[predicate] = f"{signature} -> {predicate}"
+
+        # We don't need to recompute this ever; let's just compute it once and cache it.
+        self._valid_actions: Dict[str, List[str]] = None
+
+    def is_table_entity(self, entity_name: str) -> bool:
+        """
+        Returns ``True`` if the given entity is one of the entities in the table.
+        """
+        return entity_name in self._entity_set
+
+    def _get_curried_functions(self) -> Dict[Type, int]:
```

Comment:
I think this needs an `@overrides` decorator.

Reply:
Done.